app/src/main/java/io/gnosis/safe/ui/StartActivity.kt

```diff
@@ -131,7 +131,15 @@ class StartActivity : BaseActivity(), SafeOverviewNavigationHandler, AppStateLis
                 // do not start rate flow and update screen together
                 when {
                     settingsHandler.showUpdateInfo -> askToUpdate()
-                    settingsHandler.askForPasscodeSetupOnFirstLaunch -> setupPasscode()
+                    settingsHandler.askForPasscodeSetupOnFirstLaunch -> {
+                        if (!settingsHandler.usePasscode) {
+                            setupPasscode()
+                        } else {
+                            settingsHandler.askForPasscodeSetupOnFirstLaunch = false
+                            settingsHandler.requirePasscodeToOpen = true
```

Comment:
Is that the default for upgrading users?

Reply:
That is what I meant. So if the do nothing, they need the passcode to open the app after upgrading.

Reply:
yes, as discussed in https://gnosisinc.slack.com/archives/CQB729LLF/p1621249552233600